backend/app/user_handler.go

```diff
@@ -628,18 +628,21 @@ func (h *Handler) ChargeBalance(c *gin.Context) {
 		return
 	}
 
-	user.CreditCardBalance += float64(request.Amount)
-
-	err = h.db.UpdateUserByID(&user)
+	err = internal.TransferTFTs(h.substrateClient, request.Amount, user.Mnemonic, h.config.SystemAccount.Mnemonic)
```

Comment:
Can't we precheck on the balance at this moment? (it's not guaranteed to be 100% accurate though, given there're other people can be recharging their card at that time, but at least it can give us a hint if it's going to fail with insufficient balance in the upcoming steps